### PR TITLE
Certificate addition for JupyterHub addon

### PR DESCRIPTION
**Pull request by youngjeong46**
*Issue #, if available:*
Currently to leverage ACM certificate for JupyterHub Ingress, user must manually annotate once the certificate is created (or read in using Lookup). This PR will ensure the addon will automatically apply the annotation using user provided certificate.

*Description of changes:*
Adds a configuration for user to be able to provide the name of an ACM certificate for annotation. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/jupyterhub/index.ts [hunk 1]
```diff
@@ -173,15 +180,31 @@ export class JupyterHubAddOn extends HelmAddOn {
...
+            };
+
+            if (cert){
+                presetAnnotations['alb.ingress.kubernetes.io/ssl-redirect'] = '443';
+                presetAnnotations['alb.ingress.kubernetes.io/listen-ports'] = '[{"HTTP": 80},{"HTTPS":443}]';
+                const certificate = clusterInfo.getResource<ICertificate>(cert);
+                presetAnnotations['alb.ingress.kubernetes.io/certificate-arn'] = certificate?.certificateArn;
+            } 
+
+            const annotations = { ...ingressAnnotations, ...presetAnnotations}
```
Comment (shapirov103) [inline lib/addons/jupyterhub/index.ts:200]: minor: please address github action warning on missing semicolon.

File: lib/addons/jupyterhub/index.ts [hunk 2]
```diff
@@ -173,15 +180,31 @@ export class JupyterHubAddOn extends HelmAddOn {
...
+            setPath(values, "ingress.annotations", annotations);
             setPath(values, "ingress.hosts", ingressHosts);
             setPath(values, "proxy.service", {"type" : "NodePort"});
         } else {
             assert(!ingressHosts || ingressHosts.length == 0, 'Ingress Hosts CANNOT be assigned when ingress is disabled');
             assert(!ingressAnnotations, 'Ingress annotations CANNOT be assigned when ingress is disabled');
+            assert(!cert, 'You cannot provide a certificate for NLB.');
```
Comment (shapirov103) [inline lib/addons/jupyterhub/index.ts:207]: a cert for NLB? it may be confusing since it does not look like customers specify that they need NLB, you just default to it. I would reword to state "cert option is only support if ingress is enabled" or something similar.

File: docs/addons/jupyterhub.md [hunk 3]
```diff
@@ -68,8 +88,10 @@ user-scheduler-7dbd789bc4-gcb8z   1/1     Running   0          23m
...
-4. Supports [standard helm configuration options](./index.md#standard-helm-add-on-configuration-options).
+4. If enabled, will front the JupyterHub UI with an Ingress using AWS Application Load Balancer. ***This requires AWS Load Balancer Controller add-on and it must be in add-on array before the JupyterHub add-on***. This will also look for any additional Ingress annotations provided by the user to be tagged.
+5. (Optional) Annotates Ingress with user-provided AWS Certificate Manager certificate name. It will be looked up and automatically tagged to be used with Ingress. It will require user to provide a DNS name and ***External DNS add-on to be added in add-on array before the JupyterHub add-on***.
+6. (Optional) Leverage a different notebook stack than the standard one provided. Jupyter team maintains a set of Docker image definition in a GitHub repository as explained [here](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html).
```
Comment (shapirov103) [inline docs/addons/jupyterhub.md:93]: minor: <s> Leverage </s> Leverages. Or alternatively: Customer can choose a different notebook stack ...

**Review by shapirov103 — COMMENTED** [review]
Looks great, a few minor comments

File: lib/addons/jupyterhub/index.ts (hunk 1 above)
Comment (youngjeong46) [reply lib/addons/jupyterhub/index.ts]: addressed

File: lib/addons/jupyterhub/index.ts (hunk 2 above)
Comment (youngjeong46) [reply lib/addons/jupyterhub/index.ts]: addressed

File: docs/addons/jupyterhub.md (hunk 3 above)
Comment (youngjeong46) [reply docs/addons/jupyterhub.md]: addressed

**Review by shapirov103 — APPROVED** [review]
LGTM
